### PR TITLE
workflows/scheduled-triage: add lock and support request actions

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR is meant to replace our current [`no-response`](https://github.com/Homebrew/.github/blob/master/.github/no-response.yml) workflow. I adapted the [actions/stale](https://github.com/actions/stale) action to try to match our existing configuration.

Essentially, by setting the stale label to `no response` and not providing a stale message, I'm telling the action not to mark any issues as stale in the conventional way but to continue to close issues that are stale (meaning `no repsonse`) after 14 days.

One question, should I rename the workflow to something other than `stale`? This might be a good idea, especially if (as I'm anticipating), we continue to add more actions to this.

CC: @MikeMcQuaid (by the way, I haven't forgotten about adding this to Homebrew/brew. I'm just trying to get this configuration to be more complete before I start adding it elsewhere).

### CODE REVIEW

**Comment by Rylan12** [comment]
Okay, so the [first run](https://github.com/Homebrew/homebrew-core/pull/65861/checks?check_run_id=1468076986) did nothing. There are currently [3 PRs](https://github.com/Homebrew/homebrew-core/labels/needs%20response) with the `no-response` label applied, and I would have expected https://github.com/Homebrew/homebrew-core/pull/61745 to be closed because it hasn't had a non-bot response in more than 14 days. It wasn't however.

Thinking about this more, I'm not sure we should try to coerce the stale action into doing what we want here. There are probably better actions available that will do what we want, so I'll look into it a little more.

**Comment by Rylan12** [comment]
As a summary, here are the different bots we currently have:

- [`request-info`](https://github.com/Homebrew/.github/blob/master/config.yml): I think this is working. I've seen some request-info messages appear.
- [`lock`](https://github.com/Homebrew/.github/blob/master/.github/lock.yml): I'm not sure this is working. The newest PR to have the `outdated` label applied seems like it is https://github.com/Homebrew/homebrew-core/pull/55286 which was locked back in June. As another example, https://github.com/Homebrew/homebrew-core/pull/60630 should have been locked and marked long ago because it's last update was in September. If we want to switch to an action, [Lock Threads](https://github.com/marketplace/actions/lock-threads) seems promising.
- [`no-response`](https://github.com/Homebrew/.github/blob/master/.github/no-response.yml): this isn't working as seen above. We can try to coerce the `stale` action to work here, but I'm not sure that's a great idea. I can't seem to find an action that easily does what we want here. We may be able to make our own, though, if that's desired.
- [`stale`](https://github.com/Homebrew/.github/blob/master/.github/stale.yml): Replaced already
- [`support`](https://github.com/Homebrew/.github/blob/master/.github/support.yml): It looks like this can be used as an action instead if desired.

**Review by SMillerDev — DISMISSED** [review]

**Comment by jonchang** [comment]
Thanks @Rylan12! Feel free to merge this when you're comfortable with it. 

As for renaming the workflow file, arguably it's also triaging issues. Maybe triage-daily or scheduled-triage? :shrug:

**Review by MikeMcQuaid — COMMENTED** [review]
> Essentially, by setting the stale label to `no response` and not providing a stale message, I'm telling the action not to mark any issues as stale in the conventional way but to continue to close issues that are stale (meaning `no repsonse`) after 14 days.

Ideally we'd make this so that only comments from the author avoid it being closed (i.e. comments from maintainers or other users do not reset the timer). Is that possible/happening with this action?

> * [`lock`](https://github.com/Homebrew/.github/blob/master/.github/lock.yml): I'm not sure this is working. The newest PR to have the `outdated` label applied seems like it is #55286 which was locked back in June. As another example, #60630 should have been locked and marked long ago because it's last update was in September. If we want to switch to an action, [Lock Threads](https://github.com/marketplace/actions/lock-threads) seems promising.
> * [`support`](https://github.com/Homebrew/.github/blob/master/.github/support.yml): It looks like this can be used as an action instead if desired.

This would be great.

**Review by chenrui333 — DISMISSED** [review]

**Comment by Rylan12** [comment]
> As for renaming the workflow file, arguably it's also triaging issues. Maybe triage-daily or scheduled-triage? 🤷

Leaning toward `schedules-triage`... I'll give it some thought. I was thinking  something like `repo-health-monitor` but that feels a little too... medical 😄🏥🩹🩺
 
> Ideally we'd make this so that only comments from the author avoid it being closed (i.e. comments from maintainers or other users do not reset the timer). Is that possible/happening with this action?

Yeah, that's not what's happening here. Not sure it's quite the right fit for this, so I'm not going to force it. Maybe if I've got sometime later this week I'll look into implementing our/my own action that does what we're looking for. Not sure how much effort will need to go into this so we'll just have to see (plus, I've still got to finish up the formula list migration 😅)

> > * [`lock`](https://github.com/Homebrew/.github/blob/master/.github/lock.yml): I'm not sure this is working. The newest PR to have the `outdated` label applied seems like it is #55286 which was locked back in June. As another example, #60630 should have been locked and marked long ago because it's last update was in September. If we want to switch to an action, [Lock Threads](https://github.com/marketplace/actions/lock-threads) seems promising.
> > * [`support`](https://github.com/Homebrew/.github/blob/master/.github/support.yml): It looks like this can be used as an action instead if desired.
> 
> This would be great.

Sounds good. I'll try to get those working in this PR. Once they're ready, I think I'll feel better about adding this to Homebrew/brew. The no-response feels a little less crucial and can be added later. Given that I've received some 👍 and on the general direction, I'll probably go ahead and merge one it seems to be in a state that I like.

File: .github/workflows/scheduled-triage.yml
Comment (dawidd6) [inline .github/workflows/scheduled-triage.yml:1]:
```suggestion
name: Triage issues on a schedule
```

File: .github/workflows/scheduled-triage.yml
Comment (reitermarkus) [inline .github/workflows/scheduled-triage.yml:41]: These should be pinned to a commit SHA.

**Review by reitermarkus — DISMISSED** [review]

**Comment by MikeMcQuaid** [comment]
> Leaning toward `scheduled-triage`... I'll give it some thought. I was thinking something like `repo-health-monitor` but that feels a little too... medical 😄🏥🩹🩺

Yeh, having fewer scheduled checks CI status would be nice 👍🏻 



> Yeah, that's not what's happening here. Not sure it's quite the right fit for this, so I'm not going to force it. Maybe if I've got sometime later this week I'll look into implementing our/my own action that does what we're looking for. Not sure how much effort will need to go into this so we'll just have to see (plus, I've still got to finish up the formula list migration 😅)
> Sounds good. I'll try to get those working in this PR. Once they're ready, I think I'll feel better about adding this to Homebrew/brew. The no-response feels a little less crucial and can be added later. Given that I've received some 👍 and on the general direction, I'll probably go ahead and merge one it seems to be in a state that I like.

I think it's not worth having extra effort here and we can just rely on the existing stalebot rules?



Nice work again @Rylan12!

**Comment by Rylan12** [comment]
Sorry, I'm just a little confused by the last message:

> > Leaning toward `scheduled-triage`... I'll give it some thought. I was thinking something like `repo-health-monitor` but that feels a little too... medical 😄🏥🩹🩺
> 
> Yeh, having fewer scheduled checks CI status would be nice 👍🏻

This would still be scheduled no matter what the name is, right? I might be missing what you're saying.

> > Yeah, that's not what's happening here. Not sure it's quite the right fit for this, so I'm not going to force it. Maybe if I've got sometime later this week I'll look into implementing our/my own action that does what we're looking for. Not sure how much effort will need to go into this so we'll just have to see (plus, I've still got to finish up the formula list migration 😅)
> > Sounds good. I'll try to get those working in this PR. Once they're ready, I think I'll feel better about adding this to Homebrew/brew. The no-response feels a little less crucial and can be added later. Given that I've received some 👍 and on the general direction, I'll probably go ahead and merge one it seems to be in a state that I like.
> 
> I think it's not worth having extra effort here and we can just rely on the existing stalebot rules?

I don't believe that our current lock and stale bot are working at all, so I think we can't rely on those. The support request bot _is_ working, so we don't _need_ to move to an action but I got the sense that we wanted to migrate as many of the bots to actions as we could. Can you clarify what you're thinking is best?

I just want to make sure that we're on the same page before I merge something that isn't what we want

**Comment by MikeMcQuaid** [comment]
> This would still be scheduled no matter what the name is, right? I might be missing what you're saying.

Sorry! I'm saying I think having a single scheduled workflow show up as a CI status in PRs would be ideal (rather than multiple).



> I don't believe that our current lock and stale bot are working at all, so I think we can't rely on those. The support request bot _is_ working, so we don't _need_ to move to an action but I got the sense that we wanted to migrate as many of the bots to actions as we could. Can you clarify what you're thinking is best?

Sorry, yeh, I mean:
- keep going with the stale bot as-is here
- the support bot can either stay as-is or go to an action or die (I don't mind). My take is it doesn't seem to be used much?
- lock bot would be good to find a replacement for
- "no response bot" can just be removed as the stale bot will clean up things without responses


> * [x]  Migrate existing stale bot to GitHub actions
> * [ ]  Migrate existing lock bot to GitHub actions (this PR)

Yup!

**Comment by MikeMcQuaid** [comment]
> Hmm, I still think I'm missing something (it's probably just me being relatively new to GitHub actions)
> 
> Right now, this workflow will run once a day and will mark/close stale issues and lock outdated threads. This is a "single scheduled workflow" but it doesn't "show up as a CI status in PRs" and I'm not sure what you mean by "multiple."

Sorry, it's almost entirely unrelated to this PR. My "multiple" I mean "commit style and triage in the CI results for PRs like this would ideally be a single status".



> If there's no preference I'd say just remove it. The `support-request` label has been used [7 times](https://github.com/Homebrew/homebrew-core/issues?q=label%3Asupport-request+is%3Aclosed) in homebrew/core, most recently on May 15, 2019 (over a year ago). So it doesn't sound like it will be missed.

🔥 👍🏻 



> Good point. I'll remove it

👍🏻

**Comment by Rylan12** [comment]
> Sorry, it's almost entirely unrelated to this PR. My "multiple" I mean "commit style and triage in the CI results for PRs like this would ideally be a single status".

Okay, I understand what you're saying now. Is it a goal in the future to have these checks be a CI item?

**Comment by Rylan12** [comment]
Okay, good news, the lock action seems to be working!

Here's an example of an issue that was locked: https://github.com/Homebrew/homebrew-core/pull/63880

Looking at the [list of PRs marked as `outdated`](https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+is%3Aclosed+label%3Aoutdated+) now shows what I would expect (previously, the most recent issue that was locked was in May). I'll let this sit for an hour or so just in case there are any last-minute reviews and then I'll go ahead and merge (and follow up in homebrew/brew)

**Comment by MikeMcQuaid** [comment]
Thanks @Rylan12!

**Pull request merged**
